container_restart_without_docker.md

```diff
@@ -0,0 +1,133 @@
+
+I have seen no public library functions that will perform any of this logic for
+us, which leaves us implementing it ourselves (including stuff like: base image
+has `ENV FOO=1`, child image has `ENV BAR=${FOO}2`, `ENTRYPOINT ${BAR}`).
+
+Aside from the technical risk of getting those details right, all of the
+behaviors described above are for a specific version of Docker and Kubernetes.
+I do not know if they have changed in the past or will change in the future. If
+a user runs Tilt with a version of Docker or Kubernetes that has different
+behavior, we're now potentially running the wrong command for them.
+
+While it doesn't necessarily make sense to rule out ever implementing this
+logic, it does seem pretty expensive and risky to do for what currently seems
+to be a niche case for which there is an existing workaround.
+
+Another option would be to implement only a subset of the logic, if we could
+write guards to detect and give good error messages on cases we don't yet
+support (e.g., environment variables). I'm not sure writing those guards is
+significantly cheaper than implementing the proper logic itself, but it reduces
+exposure to versioning risks (which I'm not sure are significant).
+
+#### Run the image and observe the entry point
+We could quickly run the container, grab argv for its process 1, and throw it
+away. This seems fraught:
+1. If the process has side effects, this is bad.
+2. It seems confusing to the user for this ephemeral pod/container to show up.
+3. We'd probably want to do this in a way that's as hidden from the user as
+   possible, like in a pod that doesn't have a manifest associated with it
+   so that its status / logs don't show up in the UI.
+4. If the code is in a state such that the process crashes on startup then we
+   can't get its argv this way (AFAIK) and we have a new failure mode that's
+   kinda awkward to explain to users.
+   1. Well, "process crashes on startup" is really an existing failure mode,
+      except now we're doing it in a case where the failure log isn't streamed,
+      and we're not quite using the existing pod watch flow.
+
+#### Require the user to specify the entrypoint
+OK, so we need the entrypoint but it's hard to get. Why don't we just have the
+user tell us what entrypoint to use?
+
+1. If we want Tiltfiles to be runtime-independent, this means we need to require
+   the entrypoint even for people using docker. This seems like it's making life
+   worse for the (current) majority to help out a (currently) small minority,
+   who already have a workaround available.
+2. Specifying the entrypoint in the Tiltfile is ugly duplication and an
+   opportunity for bugs and really kind of an embarrassing UI, since a user
+   would have a reasonable expectation that Tilt should be able to figure this
+   out.
+
+#### Can we just, like, not?
+1. Deprecate the `container_restart` directive
+2. Implement entry point overriding at the image level
```

Comment:
```suggestion
2. [DONE] Implement entry point overriding at the image level (this will allow users to use the same Dockerfile as they do for prod, but have it run `./wrapper.sh /my_bin` while developing)
```